malsim/sims/mal_simulator.py

```diff
@@ -654,9 +655,15 @@ def _observe_defender(
                     observation["observed_state"][index] = 1
                 else:
                     if node.is_compromised():
-                        observation["observed_state"][index] = 1
+                        fn_rate = node.extras.get('false_negative_rate')
+                        make_false_negative = fn_rate and random() < fn_rate
```

Comment:
I would advise using a random number generator object here, so that the simulator respects the given seed.

Reply:
Make sense, will fix that.

Reply:
I made a lot of changes to the PR after restructuring was done in the main branch. Needs to be reviewed again.

Reply:
I also added a test for the rng functionality to make sure a seed makes it deterministic.